.github/workflows/pre-submit.yml

```diff
@@ -24,3 +24,6 @@ jobs:
           go build -mod=vendor
           # Tests
           go test -mod=vendor -v ./...
+          # go mod tidy
+          go mod tidy
```

Comment:
why do we need to do that? We want to test the config used to build the builders. This may change the go.mod.
I think this should be a pre-submit verifying that go.mod is clean, not a command we use before building.

Did I understand the use case properly?

Reply:
OK, I understand. The `go.sum` wasn't updated so if not here where should this change be?

Even better why store the `vendor` folder in git similar to `k8s`. That will make the builds hermetic.

Reply:
Removed that go mod tidy.

Reply:
What if one of your deps gets deleted? How do you handle that?

Reply:
We don't handle it, and I think we'd use the copy from the Go module proxy if this became a problem?

Reply:
Ok.